srcalc/src/main/java/gov/va/med/srcalc/web/controller/admin/ProcedureCsvReader.java

```diff
@@ -0,0 +1,278 @@
+                    rowNumber,
+                    "cptCode",
+                    String.class,
+                    ValidationCodes.BAD_FIXED_LENGTH,
+                    new Object[] { Procedure.CPT_CODE_LENGTH },
+                    "invalid length"));
+        }
+        
+        final String complexity = tryGetValue(record, 3);
+        ValidationUtils2.validateRequiredString(
+                complexity, rowNumber, "complexity", Procedure.COMPLEXITY_MAX, ourErrors);
+        
+        final String longDesc = tryGetValue(record, 4);
+        ValidationUtils2.validateRequiredString(
+                longDesc, rowNumber, "longDescription", Procedure.DESCRIPTION_MAX, ourErrors);
+        
+        final String shortDesc = tryGetValue(record, 5);
+        ValidationUtils2.validateRequiredString(
+                shortDesc, rowNumber, "shortDescription", Procedure.DESCRIPTION_MAX, ourErrors);
+        
+        final String rvuString = tryGetValue(record, 1);
+        float rvu = Float.NaN;
+        try
+        {
+            rvu = Float.parseFloat(rvuString);
+        }
+        catch (final NumberFormatException ex)
+        {
+            ourErrors.add(TabularUploadError.forField(
+                    rowNumber,
+                    "rvu",
+                    float.class,
+                    ValidationCodes.TYPE_MISMATCH,
+                    null,
+                    ex.getMessage()));
+        }
+
+        final String eligibleString = tryGetValue(record, 2);
+        boolean eligible = false;
+        try
+        {
+            eligible = parseCsvBoolean(eligibleString);
+        }
+        catch (final IllegalArgumentException ex)
+        {
+            // Error on parseCsvBoolean()
+            ourErrors.add(TabularUploadError.forField(
+                    rowNumber,
+                    "eligible",
+                    boolean.class,
+                    ValidationCodes.TYPE_MISMATCH,
+                    null,
+                    ex.getMessage()));
+        }
+            
+        // Only create the procedure if we did not record any validation ourErrors for
+        // this record--otherwise we would get an Exception when trying to construct
+        // the Procedure.
+        if (ourErrors.isEmpty())
+        {
+            return Optional.of(new Procedure(
+                    cptCode, rvu, shortDesc, longDesc, complexity, eligible));
+        }
+
+        errors.addAll(ourErrors);
+
+        return Optional.absent();
+    }
+
+    /**
+     * Parses the given CSVRecords from a CSV file into a List of Procedures.
+     * @param records must be non-empty
+     * @param errors for recording validation errors
+     */
+    private ParseResult readProcedures(
+            final List<CSVRecord> records, final Collection<TabularUploadError> errors)
+    {
+        int firstRow = 0;
+        
+        // Detect if the first row is a header and strip if so.
+        if (records.get(0).get(0).equalsIgnoreCase("cpt"))
```

Comment:
Nothing wrong with this code, but it should be stated somewhere in documentation about what the header would look like if it were included.